python3/libexec/nbd_client_manager.py

```diff
@@ -45,7 +45,7 @@ class NbdDeviceNotFound(Exception):
 
     def __init__(self, nbd_device):
         super().__init__(
-            f"NBD device '{nbd_device}' does not exist"
+            "NBD device '{}' does not exist".format(nbd_device)
```

Comment:
f-strings were introduced in 3.6, am I missing something? https://docs.python.org/3.6/whatsnew/3.6.html#pep-498-formatted-string-literals

Reply:
You're right. Python3.6 supports f-string.
I tried running the below f-string code directly on XS8, and it works:

```
[root@lcy2-dt49 ~]# python3 -V
Python 3.6.8
[root@lcy2-dt49 ~]# python3
Python 3.6.8 (default, Nov 16 2020, 16:55:22) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on Linux
Type "help", "copyright", "credits" or "license" for more information.
>>> nbd_device = "nbd2"
>>> f"NBD device '{nbd_device}' does not exist"
"NBD device 'nbd2' does not exist"
```
Now I have the same confusion as you, why the build error? I'm not the only one who saw this error. @acefei 
Suspect that during the build process, the environment doesn't support f-string. Need to investigate further.
```
Bytecompiling .py files below /builddir/build/BUILDROOT/xapi-24.11.0.21.gcb135e00f-1.1.g2e65c85.xs8.x86_64/usr/lib/python3.6 using /usr/bin/python3.6
Compiling /builddir/build/BUILDROOT/xapi-24.11.0.21.gcb135e00f-1.1.g2e65c85.xs8.x86_64/opt/xensource/libexec/nbd_client_manager.py ...
File "/opt/xensource/libexec/nbd_client_manager.py", line 48
f"NBD device '{nbd_device}' does not exist"
^
SyntaxError: invalid syntax
```

Reply:
I encounted the similiar issue, but it failed as `py_byte_compile` running (https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_source_files_and_bytecode_cache) both py2 compile and py3 compile against .py files in the current xapi.spec. in this case, f-string won't be accepted by py2 compiling.
```
+ /usr/lib/rpm/brp-python-bytecompile /usr/bin/python2 1
Bytecompiling .py files below /builddir/build/BUILDROOT/xapi-24.11.0.92.g86c1e653a-1.4.g7cc3472.xs8.x86_64/usr/lib/python2.7 using /usr/bin/python2.7
Bytecompiling .py files below /builddir/build/BUILDROOT/xapi-24.11.0.92.g86c1e653a-1.4.g7cc3472.xs8.x86_64/usr/lib/python3.6 using /usr/bin/python3.6
```

For @stephenchengCloud 's case, it is weird that I can't reproduce it on my mock env, and I tried it as follows
```
<mock-chroot> sh-4.2$ cat > 1.py
a='123'
print(f"{a}")
<mock-chroot> sh-4.2$ python3 ./1.py
123
<mock-chroot> sh-4.2$ ls
1.py  build
<mock-chroot> sh-4.2$ python3 -m py_compile 1.py
<mock-chroot> sh-4.2$ ls
1.py  __pycache__  build
<mock-chroot> sh-4.2$ cd __pycache__/
<mock-chroot> sh-4.2$ ls
1.cpython-36.pyc
<mock-chroot> sh-4.2$ ls
1.cpython-36.pyc
```

Reply:
I think my issue is the same as yours. I managed to reproduce the issue and updated the error log on my previous comment and it's the same as yours.

I will change the PR commit message to include this info.